src/coreclr/vm/jitinterface.cpp

```diff
@@ -11777,13 +11777,13 @@ bool CEEInfo::getStaticFieldContent(CORINFO_FIELD_HANDLE fieldHnd, uint8_t* buff
     {
         if (field->IsObjRef())
         {
-            GCX_COOP();
-
-            _ASSERT(!field->IsRVA());
-            _ASSERT(valueOffset == 0); // there is no point in returning a chunk of a gc handle
-            _ASSERT((UINT)bufferSize == field->GetSize());
+            // there is no point in returning a chunk of a gc handle
+            if ((valueOffset == 0) && (field->GetSize() <= (UINT)bufferSize) && !field->IsRVA())
```

Comment:
```suggestion
            if ((valueOffset == 0) && (sizeof(CORINFO_OBJECT_HANDLE) <= (UINT)bufferSize) && !field->IsRVA())
```
`field->GetSize()` is just an expensive way to compute a constant here. We have checked that the field is IsObjRef above, so we know its size